xoa_driver/functions/anlt.py

```diff
@@ -20,191 +28,279 @@
+                    nrz_pam4_init_cond=LinkTrainingInitCondition.NO_INIT,
+                    nrz_preset=NRZPreset.NRZ_WITH_PRESET if lt_preset0_std else NRZPreset.NRZ_NO_PRESET,
+                    timeout_mode=TimeoutMode.DISABLED
+                    )
+                ]
+        else:
+            # No AN, no LT
+            tokens += [
+                commands.PP_AUTONEG(conn, mid, pid).set(
+                    AutoNegMode.ANEG_OFF,
+                    AutoNegTecAbility.DEFAULT_TECH_MODE,
+                    AutoNegFECOption.NO_FEC,
+                    AutoNegFECOption.NO_FEC,
+                    PauseMode.NO_PAUSE,
+                )
+            ]
+        await apply(*tokens)
+        
+            
+            
+
+def autoneg_timeout(
+    port: GenericAnyPort,
+    timeout: bool
+    ):
+    """Configure auto-negotiation's timeout
 
-    :param port: The port to release
+    :param port: the port to configure AN
     :type port: :class:`~xoa_driver.ports.GenericAnyPort`
+    :param timeout: whether autoneg should have timeout enabled
+    :type timeout: bool
     :return:
     :rtype: None
     """
-    await port.reservation.set_release()
-    return None
-
-
-async def anlt_status(
-    port: GenericAnyPort,
-) -> Dict[str, Any]:
-    """Get ANLT status
-
-    :param port: the port to get ANLT status from
-    :type port: :class:`~xoa_driver.ports.GenericAnyPort`
-    :return: ANLT status
-    :rtype: typing.Dict[str, Any]
-    """
-
-    # if not isinstance(port, LinkTrainingSupported):
-    #     raise NotSupportLinkTrainError(port)
     conn, mid, pid = port._conn, port.kind.module_id, port.kind.port_id
-    r0 = commands.PL1_CFG_TMP(
-        _connection=conn, _module=mid, _port=pid, _serdes_xindex=0, _type=0
-    ).get()
-    r1 = commands.PP_AUTONEGSTATUS(_connection=conn, _module=mid, _port=pid).get()
-    r2 = commands.PP_LINKTRAIN(_connection=port._conn, _module=mid, _port=pid).get()
-
-    tokens = [
-        # PL1_CFG_TMP[0,0] ?,
-        r0,
-        # port.pcs_pma.auto_neg.status.get(),
-        r1,
-        # port.pcs_pma.link_training.settings.get(),
-        r2,
+    autoneg_enabled = AutoNegMode(False)
+
+    tokens = []
+    tokens += [
+        commands.PP_LINKTRAIN(conn, mid, pid).set(
+            mode=LinkTrainingMode.DISABLED,
+            pam4_frame_size=PAM4FrameSize.P16K_FRAME,
+            nrz_pam4_init_cond=LinkTrainingInitCondition.NO_INIT,
+            nrz_preset=NRZPreset.NRZ_NO_PRESET,
+            timeout_mode=TimeoutMode.DISABLED if timeout==False else TimeoutMode.DEFAULT,
+        )
     ]
-    *_, link_recovery, autoneg, linktrain = await apply(*tokens)
-    return {
-        "auto_neg_enabled": (autoneg.mode),
-        "link_train_mode": (linktrain.mode),
-        "link_train_timeout": (linktrain.timeout_mode),
-        "link_recovery": (link_recovery.on_off),
-    }
+    # await apply(*tokens)
+    # return None
+    return tokens
```

Comment:
Function is not working.  `commands.PP_LINKTRAIN(...).set(...)` is never being awaited